doc/code/qml_grouping.rst

```diff
@@ -0,0 +1,93 @@
+qml.grouping
+============
+
+This subpackage defines functions and classes for Pauli word partitioning
+functionality used in measurement optimization.
+
+A Pauli word is defined as :math:`P_I = \prod_{i=1}^{N}\sigma_i^{(I)}`, where
+:math:`\sigma_i^{(I)}` is one of the Pauli operators (:class:`~pennylane.PauliX`,
+:class:`~pennylane.PauliY`, :class:`~pennylane.PauliZ`) or identity
+(:class:`~pennylane.Identity`) acting on the :math:`i^{th}` qubit.
+
+Pauli words can be used for expressing a qubit :class:`~pennylane.Hamiltonian`.
+A qubit Hamiltonian has the form :math:`H_{q} = \sum_{I} C_I P_I` where
+:math:`C_{I}` are numerical coefficients, and :math:`P_I` are Pauli words.

```

Comment:
Can we use an index set other than `I`? Might be some possibility of confusing it with Identity operator?

Reply:
Good idea! Updated with `J`